### PR TITLE
[Feature]Add operational state cluster xml and code gen

### PR DESCRIPTION
**Pull request by mideayanghui**
Seperate the operational state cluster XML to a new PR. Based on `[New XML and Code Gen for Appliance Clusters (OpState, TCC, RefAlm #26560 )](https://github.com/project-chip/connectedhomeip/pull/26560#top)` 

Problem:
Matter 1.2 clusters missing XML and generated code for operational state cluster

Solution:
Create XML
Update zcl.json, zcl-with-test-extensions.json and related zap files.

Testing:
Regen all
Rebuild chip tool
Verify all new clusters are visible in chip-tool, with correct set of available attributes

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/operational-state-cluster.xml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    <cluster code="0x0060"/>
+    <item name="OperationalStateID"    type="OperationalStateEnum" optional="false"/>
+    <item name="OperationalStateLabel" type="CHAR_STRING" optional="false"/>
+  </struct>
+
+  <struct name="ErrorStateStruct">
+    <cluster code="0x0060"/>
+    <item name="ErrorStateID" type="ErrorStateEnum" optional="false" />
+    <item name="ErrorStateLabel" type="CHAR_STRING" optional="false" length="64"/>
+    <item name="ErrorStateDetails" type="CHAR_STRING" optional="false" length="64"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/operational-state-cluster.xml:46]: ErrorStateLabel is supposed to be nullable and ErrorStateDetails is supposed to be optional, no?
```suggestion
    <item name="ErrorStateLabel" type="CHAR_STRING" optional="false" isNullable="true" length="64"/>
    <item name="ErrorStateDetails" type="CHAR_STRING" optional="true" length="64"/>
```

File: src/app/zap-templates/zcl/data-model/chip/operational-state-cluster.xml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    <define>OPERATIONAL_STATE_CLUSTER</define>
+    <client init="false" tick="false">true</client>
+    <server init="false" tick="false">true</server>
+    <description>This cluster supports remotely monitoring and, where supported, changing the operational state of any device where a state machine is a part of the operation.</description>
+
+    <globalAttribute side="server" code="0xFFFD" value="1" />
+
+    <attribute side="server" code="0x0000" define="PHASE_LIST"              type="ARRAY" entryType="CHAR_STRING"                                     writable="false" isNullable="true"  optional="false">PhaseList</attribute>
+    <attribute side="server" code="0x0001" define="CURRENT_PHASE"           type="INT8U"                                    min="0x00" max="0x1F"    writable="false" isNullable="true"  optional="false">CurrentPhase</attribute>
+    <attribute side="server" code="0x0002" define="COUNTDOWN_TIME"          type="INT32U"                                min="0x00" max="0x3F480" writable="false" isNullable="true"  optional="true">CountdownTime</attribute>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/operational-state-cluster.xml:62]:
```suggestion
    <attribute side="server" code="0x0002" define="COUNTDOWN_TIME"          type="elapsed_s"                                min="0x00" max="259200" writable="false" isNullable="true"  optional="true">CountdownTime</attribute>
```
to more clearly match the spec.

File: src/app/zap-templates/zcl/data-model/chip/operational-state-cluster.xml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    <attribute side="server" code="0x0003" define="OPERATIONAL_STATE_LIST"  type="ARRAY" entryType="OperationalStateStruct"                          writable="false"                    optional="false">OperationalStateList</attribute>
+    <attribute side="server" code="0x0004" define="OPERATIONAL_STATE"       type="OperationalStateStruct"                                            writable="false"                    optional="false">OperationalState</attribute>
+    <attribute side="server" code="0x0005" define="OPERATIONAL_ERROR"       type="ErrorStateStruct"                                                   writable="false"                    optional="false">OperationalError</attribute>
+
+    <command source="client" code="0x00" name="Pause" response="OperationalCommandResponse" optional="false">
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/operational-state-cluster.xml:67]: All the commands here are marked optional in the spec, no?

File: src/app/zap-templates/zcl/data-model/chip/operational-state-cluster.xml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    <event side="server" code="0x00" priority="critical" name="OperationalError" optional="false">
+      <description>OperationalError</description>
+      <field id="0" name="ErrorState" type="ErrorStateStruct" />
+    </event>
+
+     <event side="server" code="0x01" priority="info" name="OperationCompletion" optional="true">
+      <description>OperationCompletion</description>
+      <field id="0" name="CompletionErrorCode"  type="ErrorStateEnum" />
+      <field id="1" name="TotalOperationalTime" type="INT32U" />
+      <field id="2" name="PausedTime"           type="INT32U" />
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/operational-state-cluster.xml:97]: Both of these fields are supposed to be optional _and_ nullable, as far as I can tell.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src/app/zap-templates/zcl/data-model/chip/operational-state-cluster.xml (hunk 3 above)
Comment (mideayanghui) [reply src/app/zap-templates/zcl/data-model/chip/operational-state-cluster.xml:67]: It seems that OPERATIONAL_STATE_LIST、OPERATIONAL_STATE、OPERATIONAL_ERROR are mandatory from https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/app_clusters/OperationalState.adoc

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
@mideayanghui There seem to be java codegen bits missing?

**Comment by mideayanghui** [comment]
> @mideayanghui There seem to be java codegen bits missing?

Do you have any suggestions for fix this bug of java codegen? Thanks.

**Comment by bzbarsky-apple** [comment]
> Do you have any suggestions for fix this bug of java codegen? Thanks.

It's _possible_ that running regen_all twice will fix things.  @andy31415

**Pull request merged**